### PR TITLE
NFDIV-2632 - Email validation for online cases

### PR DESCRIPTION
**Pull request by amoriartyMOJ**
### Change description ###

Covers email validation for the following:

Applicant1 email
Applicant1 solicitor email
Applicant2 email
Applicant2 solicitor email

Example
image

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2632

**Before merging a pull request make sure that:**

- [X] tests have been updated / new tests has been added (if needed)
- [X] README and other documentation has been updated / added (if needed)

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant1Test.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                .email("invalidEmail")
+            .build());
+
+        final CaseDetails<CaseData, State> details = new CaseDetails<>();
+        details.setData(caseData);
+        details.setId(TEST_CASE_ID);
+
+        AboutToStartOrSubmitResponse<CaseData, State> response = page.midEvent(details, details);
+
+        assertEquals(response.getErrors().size(), 1);
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant1Test.java:36]: worth asserting error message

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java [hunk 2]
```diff
@@ -94,4 +107,30 @@ public void addTo(final PageBuilder pageBuilder) {
...
     }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State> details,
+                                                                  CaseDetails<CaseData, State> detailsBefore) {
+        CaseData caseData = details.getData();
+        Applicant applicant2 = caseData.getApplicant2();
+        List<String> validationErrors = new ArrayList<>();
+
+        boolean validApplicantEmail;
+        if (applicant2.getEmail() != null && !applicant2.getEmail().isEmpty()) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java:118]: worth using `import static org.apache.commons.lang3.StringUtils.isNotEmpty;`

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java [hunk 3]
```diff
@@ -94,4 +107,30 @@ public void addTo(final PageBuilder pageBuilder) {
...
+            validApplicantEmail = EmailValidator.getInstance().isValid(applicant2.getEmail());
+            if (!validApplicantEmail) {
+                validationErrors.add(INVALID_APPLICANT_EMAIL_ERROR);
+            }
+        }
+
+        if (applicant2.isRepresented()) {
+            boolean validSolicitorEmail = EmailValidator.getInstance().isValid(applicant2.getSolicitor().getEmail());
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java:126]: applicant 2 sol email is optional ?

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java (hunk 2 above)
Comment (amoriartyMOJ) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java]: Added this.

File: src/test/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant1Test.java (hunk 1 above)
Comment (amoriartyMOJ) [reply src/test/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant1Test.java:37]: Added this.

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java (hunk 3 above)
Comment (amoriartyMOJ) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java:127]: App2 sol email is only requested (mandatory) if app2 is represented.

**Review by nitinprabhu — APPROVED** [review]

**Pull request closed without merging**